### PR TITLE
Refactor/dataverse contract

### PR DESCRIPTION
**Pull request by ccamel**
This PR introduces a paradigm shift in the management of resources within the Dataverse, transitioning from a Metadata-centric approach to a Claims-based approach. According to this new framework, all resources with a decentralized identity (e.g. zones, services, digital resources, entites...) can be described through properties asserted about them by an entity. This mechanism offers greater flexibility and standardization, enhancing trust within the Dataverse.

Accordingly, the [underlying ontology](https://github.com/okp4/ontology) that supports these principles will be updated to reflect these changes.

## Summary by CodeRabbit

- **Documentation**
  - Updated the README and documentation to provide detailed explanations of the Dataverse, Zones, Resources, Decentralized Identifiers (DID), and Claims within the smart contract context.
  - Refined descriptions and examples for various contract operations such as service registration, digital resource registration, zone founding, and claim submission and revocation.

- **Refactor**
  - Simplified the smart contract interface by removing metadata attachment/detachment functions and focusing on claim submission and revocation.
  - Replaced the `Iri` type with the `Uri` type for consistency and clarity in the contract's messaging system.

### CODE REVIEW

File: contracts/okp4-dataverse/src/msg.rs [hunk 1]
```diff
@@ -45,31 +40,31 @@ pub enum ExecuteMsg {
...
+        ///
+        /// Preconditions:
+        /// - The URI must be unique within the dataverse.
         identifier: Uri,
         /// The URI of the service, already registered in the dataverse, that provides the dataset.
+        ///
+        /// Preconditions:
+        /// - The Service must be registered in the dataverse before the dataset can be registered.
         provided_by: Uri,
         /// The URI of the entity responsible for registering and managing the dataset in the dataverse (i.e. on the blockchain).
```
Comment (amimart) [inline contracts/okp4-dataverse/src/msg.rs:69]:
```suggestion
        /// The URI of the entity responsible for registering and managing the resource in the dataverse (i.e. on the blockchain).
```

File: contracts/okp4-dataverse/src/msg.rs [hunk 2]
```diff
@@ -45,31 +40,31 @@ pub enum ExecuteMsg {
...
+    /// and the specific service responsible for its provision. This dual-component identification mechanism guarantees the distinct
+    /// recognition and operationalization of each Digital Resource within the dataverse environment.
+    RegisterDigitalResource {
+        /// The decentralized identity (DID) of the Digital Resource.
+        ///
+        /// Preconditions:
+        /// - The identity must be unique within the dataverse.
+        identity: Did,
+        /// The URI that identifies the dataset.
+        /// This URI makes sense only in the context of the service that provides the dataset.
```
Comment (amimart) [inline contracts/okp4-dataverse/src/msg.rs:59]:
```suggestion
        /// This URI makes sense only in the context of the service that provides the resource.
```

File: contracts/okp4-dataverse/src/msg.rs [hunk 3]
```diff
@@ -45,31 +40,31 @@ pub enum ExecuteMsg {
...
+        ///
+        /// Preconditions:
+        /// - The URI must be unique within the dataverse.
         identifier: Uri,
         /// The URI of the service, already registered in the dataverse, that provides the dataset.
+        ///
+        /// Preconditions:
+        /// - The Service must be registered in the dataverse before the dataset can be registered.
```
Comment (amimart) [inline contracts/okp4-dataverse/src/msg.rs:67]:
```suggestion
        /// - The Service must be registered in the dataverse before the resource can be registered.
```

**Review by amimart — CHANGES_REQUESTED** [review]
I like the idea, I just noticed some documentation inconsistencies due to the renaming of `RegisterDataset` in `RegisterDigitalResource` :)

**Comment by ccamel** [comment]
@coderabbitai pause

**Review by amimart — APPROVED** [review]

**Pull request merged**